src/native/eventpipe/ep-buffer-manager.c

```diff
@@ -399,8 +404,9 @@ buffer_manager_init_sequence_point_thread_list (
 		// underflow.
 		uint32_t sequence_number = ep_thread_session_state_get_volatile_sequence_number (thread_session_state) - 1;
 
-		dn_umap_ptr_uint32_insert (ep_sequence_point_get_thread_sequence_numbers (sequence_point), thread_session_state, sequence_number);
-		ep_thread_addref (ep_thread_holder_get_thread (ep_thread_session_state_get_thread_holder_ref (thread_session_state)));
+		ep_rt_thread_id_t os_thread_id = ep_thread_get_os_thread_id (ep_thread_session_state_get_thread (thread_session_state));
+
+		dn_umap_threadid_uint32_insert (ep_sequence_point_get_thread_sequence_numbers (sequence_point), os_thread_id, sequence_number);
```

Comment:
What happens if a thread might die when being on this map? If that happens, the old code would at least preserve the internal representation of TSS through the addref and then access it through that data structure as a representation of the thread, but now when we use thread id as key, that could be reused by OS, so if everything related to the thread TSS has been freed while we hold the thread id -> sequence point representation in this map, then there might be duplicates. In theory I believe this could happen in the past as well, but we would end up with different TSS instance but with the same internal thread id, so not sure what consequences that would introduce later, just wanted to highlight this potential scenario when we introduce thread id is identification instead of TSS.